### PR TITLE
Update examples and documentation

### PR DESCRIPTION
**Pull request by msau42**

### CODE REVIEW

**Comment by msau42** [comment]
@verult and @jingxu97 can you followup and improve the documentation for snapshot and topology alpha features?  We should create a new alpha driver deployment config like I suggested [here](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/issues/212#issuecomment-449157580)

File: README.md [hunk 1]
```diff
@@ -100,6 +100,25 @@ NAME                      READY     STATUS    RESTARTS   AGE
 web-server                1/1       Running   0          1m
 ```
 
+## Kubernetes User Guide (Alpha features)
+
+### FIXME: Install Driver with alpha features
```
[the remark this answers is not in the document]
Comment (msau42) [reply README.md]: A note to fix later.  We should add a new "alpha" driver deployment overlay

File: docs/development.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+## Kubernetes Development
```
Comment (davidz627) [inline docs/development.md]: lets link to this file from the readme

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidz627) [reply README.md]: up to you whether you want to do this now or leave it for later.

But I imagined some sort of "user guides" folder which contains an `md` file for each feature/guide. The README could then link to those individually. Just wanted to let you know that's what I was thinking for when the number of user guides continues to increase

[the remark this answers is not in the document]
Comment (msau42) [reply README.md]: I like that idea, will restructure

File: docs/kubernetes/user-guides/snapshots.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+$ PROJECT=your-project-here                       # GCP project
+$ GCE_PD_SA_NAME=my-gce-pd-csi-sa                 # Name of the service account to create
+$ GCE_PD_SA_DIR=/my/safe/credentials/directory    # Directory to save the service account key
+$ ./deploy/setup-project.sh
+```
+
+2. Deploy driver to Kubernetes Cluster
+```
+$ GCE_PD_SA_DIR=/my/safe/credentials/directory    # Directory to get the service account key
+$ GCE_PD_DRIVER_VERSION=alpha                     # Driver version to deploy
```
Comment (davidz627) [inline docs/kubernetes/user-guides/snapshots.md]: are we creating a new "alpha" overlay? Right now I think `dev` basically serves this function(ish)

Comment (davidz627) [reply docs/kubernetes/user-guides/snapshots.md]: I think we should update the overlay/docs atomically so we don't have a period of time when docs say to deploy alpha but that doesn't actually do anything

Comment (msau42) [reply docs/kubernetes/user-guides/snapshots.md]: yes, i want to create a separate alpha overlay that we can use for another CI test job.  I will work on adding one

**Comment by msau42** [comment]
/hold
until an alpha overlay is created

**Comment by msau42** [comment]
I'm going to wait until 1.14 is released to test and merge this.

File: docs/kubernetes/user-guides/snapshots.md (hunk 3 above)
Comment (msau42) [reply docs/kubernetes/user-guides/snapshots.md]: I decided to just put the overlay back to dev for now.  With topology going beta in 1.14, it's only snapshots that is still alpha.

**Comment by msau42** [comment]
Updated, but still keep hold to wait until we're closer to a release

**Comment by msau42** [comment]
/hold cancel
I removed adding v0.5.0 to our official release table for now so we can get the updated docs in

File: README.md [hunk 4]
```diff
@@ -41,145 +41,35 @@ See Github [Issues](https://github.com/kubernetes-sigs/gcp-compute-persistent-di
...
-```
-
-To modify dependencies or versions change `./Gopkg.toml`
+### Kubernetes Beta Features
+
+* Topology: Requires K8s 1.14+ on Master and Nodes and PD driver v0.5.0+
+
+### Kubernetes Alpha Features
+
+* Snapshots: Requires K8s 1.13+ on Master and PD driver v0.3.0+
```
Comment (davidz627) [inline README.md]: and to use the `alpha` overlay?

File: README.md [hunk 5]
```diff
@@ -41,145 +41,35 @@ See Github [Issues](https://github.com/kubernetes-sigs/gcp-compute-persistent-di
...
-
-Use [dep](https://github.com/golang/dep)
-```
-$ dep ensure
-```
-
-To modify dependencies or versions change `./Gopkg.toml`
+### Kubernetes Beta Features
+
+* Topology: Requires K8s 1.14+ on Master and Nodes and PD driver v0.5.0+
```
Comment (davidz627) [inline README.md]: 0.5.0 not yet release

File: docs/kubernetes/user-guides/basic.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
+# Kubernetes Basic User Guide
+This guide gives a simple example on how to provision zonal and regional PDs in single-zone and regional clusters.
+
+**Note:** Regional cluster support only available in beta starting with
```
Comment (davidz627) [inline docs/kubernetes/user-guides/basic.md]: maybe move/duplicate this note in the "regional PD" section

File: README.md (hunk 5 above)
Comment (msau42) [reply README.md]: If I change it to "dev", are we going to remember to change it back?

Comment (davidz627) [reply README.md]: probably not, lets leave it like this

File: docs/kubernetes/user-guides/basic.md (hunk 6 above)
Comment (msau42) [reply docs/kubernetes/user-guides/basic.md]: I duplicated it because you can use single-zone pds in a regional cluster too

**Pull request merged**